src/Native/Unix/System.Native/pal_process.cpp

```diff
@@ -212,10 +223,16 @@ extern "C" int32_t SystemNative_ForkAndExecProcess(const char* filename,
     CloseIfOpen(waitForChildToExecPipe[WRITE_END_OF_PIPE]);
     if (waitForChildToExecPipe[READ_END_OF_PIPE] != -1)
     {
-        int ignored;
+        int childError;
         if (success)
         {
-            while (CheckInterrupted(read(waitForChildToExecPipe[READ_END_OF_PIPE], &ignored, 1)));
+            ssize_t result;
+            while (CheckInterrupted(result = read(waitForChildToExecPipe[READ_END_OF_PIPE], &childError, sizeof(childError))));
+            if (result == sizeof(childError))
+            {
+                success = false;
+                priorErrno = childError;
+            }
```

Comment:
Makes sense.  Are there any problematic situations where this would end up throwing now on Unix where it wouldn't on Windows?  Logically the execve call is part of the start operation, so presumably not.

Reply:
Indeed. This will bring us closer to Windows behavior where CreateProcess also does the actions (and reports errors for) the things that are done after the fork.